script/update_marriage_abroad_opposite_sex_outcomes.rb

```diff
@@ -0,0 +1,130 @@
+require "csv"
+require "find"
+
+# Variations
+# 1 - commission text with link
+# 2 - embassy text with link
+# 3 - government text
+# 4 - commission text with link (above ##Prove you're free to get married)
+# 5 - embassy text with link (above ##Prove you're free to get married)
+# 6 - embassy text with link (above ##What you need)
+# 7 - consulate text with link (above ##What you need to do)
+
+COUNTRIES_DIR = "lib/smart_answer_flows/marriage-abroad/outcomes/countries/".freeze
+OPPOSITE_SEX_FILE = "_opposite_sex.govspeak.erb".freeze
+CSV_HEADINGS = %w[country variation embassy_link updated].freeze
+
+def get_country_folder(country)
+  if country.include?("of the")
+    country.gsub(" ", "-").downcase
+  else
+    country.gsub("the ", "").gsub(" ", "-").downcase
+  end
+end
+
+def text_variation(variation, link)
+  case variation
+  when 1, 4 # commission text with link, eof
+    "You might be able to form an opposite sex civil partnership. Check the local rules with the [British high commission](#{link})."
+  when 2, 5, 6 # embassy text with link
+    "You might be able to form an opposite sex civil partnership. Check the local rules with the [British embassy](#{link})."
+  when 3 # government text
+    "You might be able to form an opposite sex civil partnership. Check the rules with the local government."
+  when 7 # consulate text with link
+    "You might be able to form an opposite sex civil partnership. Check the local rules with the [British consulate](#{link})."
+  end
+end
+
+def append_to_eof(opposite_sex_paths, text)
+  text = "\n" + text + "\n"
+  puts "The following text will be added to the end of files:"
+  puts text
+
+  opposite_sex_paths.each do |path|
+    File.write(path, text, mode: "a")
+    puts "File has been updated: #{path}"
+  end
+end
+
+def insert_above_heading(opposite_sex_paths, variation, text)
+  if variation.between?(4, 5)
+    heading_match = "Prove you’re free to get married"
+  elsif variation.between?(6, 7)
+    heading_match = "What you need to do"
+  end
+
+  text += "\n\n"
+  puts "The following text will be inserted above the heading ## #{heading_match}:"
+  puts text
+
+  text += "## #{heading_match}"
+
+  opposite_sex_paths.each do |path|
+    file = File.read(path)
+
+    if file.include?(heading_match)
+      if file.include?("##" + heading_match)
+        file = file.gsub("##" + heading_match, text)
+      elsif file.include?("## " + heading_match)
+        file = file.gsub("## " + heading_match, text)
+      else
+        puts "Match is not a H2 heading"
+      end
+      File.write(path, file)
+      puts "File has been updated: #{path}"
+    else
+      puts "Heading match not found in file: #{path}"
+    end
+  end
+end
+
+update_outcomes_csv = "script/update_outcomes.csv"
```

Comment:
I'd be tempted to move this to a constant, where all the other file paths are defined. Just feels neater.

Reply:
Agreed, have updated this now!